### ISSUE TITLE
Warning Section/Article lacks header when heading element is used

### ISSUE THREAD

**Issue by yesurbius** [validator/validator#468]
The validator complains

> Warning: Article lacks heading. Consider using h2-h6 elements to add identifying headings to all articles.  From line 15, column 7; to line 15, column 15

and

> Warning: Section lacks heading. Consider using h2-h6 elements to add identifying headings to all sections.

when I attempt to validate the markup (included at the bottom of post).

From the latest HTML proposed standard

> A header element is intended to usually contain the section’s heading (an h1–h6 element), but this is not required. The header element can also be used to wrap a section’s table of contents, a search form, or any relevant logos.
-- http://w3c.github.io/html/sections.html#elementdef-header

I acknowledge that the validator has only indicated a _warning_ and not an error, but when a `header` element has been explicitly used as the first child to a sectioning element (`article` or a `section`), I feel as though the warning should be suppressed as it now becomes superfluous.

```html
<!DOCTYPE html>
<html lang="en">
  <head>
    <title>Validator Warning Example</title>
  </head>
  <body>
    <header>
      <p>This is the document header</p>
    </header>
    <section>
      <header>
        <p>This is the section header</p>
      </header>
      <p>This is the section text</p>
      <article>
        <header>
          <p>This is the article header</p>
        </header>
        <p>This is the article text</p>
      </article>
    </section>
  </body>
</html>
```

**Comment by sideshowbarker** [comment]
A `header` on its own does not provide a heading.

One of the reasons for that warning is that when you put a `section` or `article` into a document and don’t put a `h1`-`h6` heading on it, then it doesn’t get exposed to users of screen readers in a way the facilitates then navigating to the section.

So markup like this:

```html
      <article>
        <header>
          <p>This is the article header</p>
        </header>
        <p>This is the article text</p>
      </article>
```
…does not expose that _This is the article header_ text as a proper heading in any way. That kind of markup case is exactly the kind of case the warning in the checker is intended to catch.

But as you note, the checker message here is a warning, not an error. If you have some reason for marking things up that way that you’re satisfied with, you can safely ignore the warning (or use the **Message filtering** button in the checker UI to persistently filter it so you won’t ever see it again).

**Comment by yesurbius** [comment]
Okay.  Thanks
